File: backend/api/src/main/kotlin/fima/api/investment/Stock.kt
package fima.api.investment

import fima.api.utils.FromProtoConvertable
import fima.domain.investment.Stock as ProtoStock

data class Stock(val symbol: String, val name: String, val index: String, val sector: String, val investmentType: String) {
    companion object : FromProtoConvertable<ProtoStock, Stock> {
        override fun fromProto(proto: ProtoStock): Stock {
            return Stock(proto.symbol, proto.name, proto.index.name, proto.sector.name, proto.investmentType.name)
        }
    }
}

File: backend/investment-service/src/main/scala/fima/services/investment/model/Stock.scala
package fima.services.investment.model

import fima.domain.investment.InvestmentDomain.{Stock => PStock}
import fima.services.investment.model.Stock.StockSymbol

import java.time.Instant

object Stock {
  type StockSymbol = String
}

case class Stock(
  symbol: StockSymbol,
  name: String,
  index: MarketIndex,
  sector: SectorType,
  investmentType: InvestmentMethod,
  marketValue: BigDecimal,
  updatedAt: Instant
) {

  def stockToProto: PStock = {
    import implicits._

    PStock(symbol, name, index, toProto(sector), investmentType)
  }

}

case class Position(
  stock: Stock,
  shares: Int
)